### PR TITLE
Make base URL configurable

### PR DESCRIPTION
**Pull request by pluehne**
This adds a command-line interface that allows users to specify an alternative base URL for the GraphQL API. This makes it possible to retrieve GraphQL schemas of GitHub Enterprise servers and not just github.com.

For the command-line interface, I used `yargs`, which is [used by a couple of other repositories in the Octokit organization](https://github.com/search?q=org%3Aoctokit+yargs&type=Code).

I’m happy to make changes as necessary, so please let me know what you think :slightly_smiling_face:.

### CODE REVIEW

**Comment by gr2m** [comment]
Hi Patrick, thanks for the pull request!

How do you intend to use this repository? So far, the download script is only used internally, we only expose the downloaded schema. Downloading schemas for Enterprise versions is something we want to add, but we don’t have the instances setup yet

**Comment by pluehne** [comment]
All right, I wasn’t aware that this repository was for internal use so far :slightly_smiling_face:.

My use case is the following. I’m working with GitHub Enterprise servers every day, and I’d like to write tools performing API queries in Rust. The GraphQL API seems to be a good choice for that purpose given that it provides access to a machine-readable specification/schema (which is not the case with the v3 SQL API as far as I understood). Next, I’d like to give the schema to [`graphql-rust`](https://github.com/graphql-rust/graphql-client) and then build my tools around that with the benefit of having type-safety in requests and responses.

Long story short, I found `bin/download.js` very convenient for my use case of accessing the GraphQL schema of a GitHub server—but it lacked the capability of specifying alternative base URLs. Hence the pull request :slightly_smiling_face:.

**Comment by gr2m** [comment]
Maybe the best approach would be to copy `bin/download.js` into your own script for the time being and make whatever changes are necessary?

**Comment by pluehne** [comment]
Sure, that’s what I did. I just thought it might be nice to open a pull request to contribute back in case you’re interested. If you want to tackle the GitHub Enterprise API in a different way than what I suggested here, that’s totally fine with me :slightly_smiling_face:. Don’t hesitate to close this pull request if it doesn’t suit your needs.

**Comment by gr2m** [comment]
Thanks Patrick. I’ll close it to now, but it’s great to have it once we have the GHE setup :)

**Comment by pluehne** [comment]
Sure! Feel free to steal from this pull request later, redo it differently, or leave it be. Whatever works best for you :slightly_smiling_face:.
